frontend/src/compononents/CocktailRecipe.tsx

```diff
@@ -2,41 +2,40 @@ import './CocktailRecipe.css';
 import {Cocktail} from "../model/Cocktail";
 import * as React from "react";
 import {Button} from "@mui/material";
-
+import {Favourite} from "../model/Favourite";
 
 type CocktailRecipeProps = {
-    cocktail : Cocktail
+    cocktail: Cocktail
+    addFavourite: (favourite: Favourite) => void
 }
 
-export default function CocktailRecipe( { cocktail } : CocktailRecipeProps ){
-
+export default function CocktailRecipe( { cocktail,addFavourite} : CocktailRecipeProps ){
 
     return <div className="cocktail-recipe_" >
         <div className="cocktail-recipe_card" >
-
         <div className="wrapper">
-            <div className="box a">{cocktail.name}<Button variant="text">Add to favorites</Button></div>
+            <div className="box a">{cocktail.name}<Button variant="text" onClick={()=> addFavourite(cocktail)}>Add to favorites</Button></div>
             <div className="box b"><img className="cocktail-recipe_image" src={cocktail.imageUrl} alt={cocktail.name}/></div>
             <div className="box c"><h3>Glass type:</h3> {cocktail.glass}</div>
-            <div className="box d"><h3>Ingredients:</h3> {cocktail.measure1} {cocktail.ingredient1}
-            <br/>{cocktail.measure2} {cocktail.ingredient2}<br/>
-                {cocktail.measure3} {cocktail.ingredient3}<br/>
-                {cocktail.measure4} {cocktail.ingredient4}<br/>
-                {cocktail.measure5} {cocktail.ingredient5}<br/>
-                {cocktail.measure6} {cocktail.ingredient6}<br/>
-                {cocktail.measure7} {cocktail.ingredient7}<br/>
-                {cocktail.measure8} {cocktail.ingredient8}<br/>
-                {cocktail.measure9} {cocktail.ingredient9}<br/>
-                {cocktail.measure10} {cocktail.ingredient10}<br/>
-                {cocktail.measure11} {cocktail.ingredient11}<br/>
-                {cocktail.measure12} {cocktail.ingredient12}<br/>
-                {cocktail.measure13} {cocktail.ingredient13}<br/>
-                {cocktail.measure14} {cocktail.ingredient14}<br/>
-                {cocktail.measure15} {cocktail.ingredient15}<br/>
+            <div className="box d"><h3>Ingredients:</h3> {cocktail.measure1} {cocktail.ingredient1}<br/>
```

Comment:
Die große Anzahl an Properties, die Dir die externe API vorgibt, ist wirklich unpraktisch.

Mit Spezialkonstrukten kann man in JavaScript trotzdem den Code reduzuieren.

Dies macht den Code kürzer, aber leider auch komplizierter. Ich bin mir deshalb unsicher, welcher Code für Dich besser ist.

Aber damit Du weißt, wie ich es meine:

Oben in der Funktion so:
```
const allIngredients = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15].map(index => {
    // @ts-ignore
    const measure = cocktail["measure" + index];
    // @ts-ignore
    const ingredient = cocktail["ingredient" + index];
    if (measure === null || measure === undefined) return;
    if (ingredient === null || ingredient === undefined) return;
    return <p>  {measure + " " + ingredient} <br/></p>
});
```

Und unten im Return so:
```
<div className="box d"><h3>Ingredients:</h3> {allIngredients}</div>
```